controller/Review.php

```diff
@@ -52,19 +57,27 @@ public function index()
         /** @var LtiResultAliasStorage $ltiResultIdStorage */
         $ltiResultIdStorage = $this->getServiceLocator()->get(LtiResultAliasStorage::SERVICE_ID);
 
-
         $resultIdentifier = $launchData->hasVariable('lis_result_sourcedid')
             ? $launchData->getVariable('lis_result_sourcedid')
             : $launchDataService->findDeliveryExecutionFromLaunchData($launchData);
 
         $deliveryExecutionId = $ltiResultIdStorage->getDeliveryExecutionId($resultIdentifier);
 
+        if ($deliveryExecutionId === null) {
+            throw new LtiInvalidLaunchDataException('Wrong result ID provided');
+        }
+
+        $execution = $this->getResource($deliveryExecutionId);
+        $delivery = $execution->getOnePropertyValue($this->getProperty(OntologyDeliveryExecution::PROPERTY_DELIVERY));
+
         if ($deliveryExecutionId !== null) {
-            $params['delivery_execution'] = $deliveryExecutionId;
+            $data = [
+                'delivery' => $delivery->getUri(),
+                'execution' => $execution->getUri()
+            ];
         }
 
-        $this->setClientRoute(_url('index', 'Review', 'taoReview', $params ?? []));
-        $this->composeView('delegated-view', null, 'pages/index.tpl', 'tao');
+        $this->composeView('delegated-view', $data ?? [], 'pages/index.tpl', 'tao');
```

Comment:
`$data ?? []` php 7.0?

Reply:
Yes, it's a new env, and we officially now supports 7.1 by Joel.